POEApi.Model.Tests/PoeModelTests.cs

```diff
@@ -125,5 +125,27 @@ public void GetAccountNameTest()
                 Assert.AreEqual(account, "fakeAccountName");
             }
         }
+
+        [TestMethod]
+        public void GetShardCurrencyStashTest()
+        {
+            string fakeStashInfo = Encoding.UTF8.GetString(Files.SampleCurrencyTabWithShards);
+            using (var stream = GenerateStreamFromString(fakeStashInfo))
+            {
+                _mockTransport.Setup(m => m.GetStash(0, "", "", false)).Returns(stream);
+
+                var stash = _model.GetStash(0, "", "");
+
+                Assert.IsNotNull(stash);
+
+                Assert.AreEqual(stash.Tabs.Count, 26);
+
+                var items = stash.GetItemsByTab(0);
+
+                Assert.AreEqual(items.OfType<Currency>().Count(x => x.Type == OrbType.EngineersOrb), 1);

```

Comment:
Might want to look into FluentAssertions for these tests, provides a nicer fluent syntax for assertions.